### PR TITLE
vio for windows: Use more graceful create file params.

### PR DESCRIPTION
**Pull request by dragotin**
shouldn't we relax the attributes a bit?

### CODE REVIEW

File: csync/src/vio/csync_vio_local_win.c [hunk 1]
```diff
@@ -230,8 +230,11 @@ int csync_vio_local_stat(const char *uri, csync_vio_file_stat_t *buf) {
...
     mbchar_t *wuri = c_utf8_path_to_locale( uri );
 
-    h = CreateFileW( wuri, 0, FILE_SHARE_READ, NULL, OPEN_EXISTING,
-                     FILE_ATTRIBUTE_NORMAL+FILE_FLAG_BACKUP_SEMANTICS+FILE_FLAG_OPEN_REPARSE_POINT, NULL );
+    h = CreateFileW( wuri, 0, FILE_SHARE_WRITE | FILE_SHARE_READ | FILE_SHARE_DELETE,
+                     NULL, OPEN_EXISTING,
+                     FILE_ATTRIBUTE_NORMAL+FILE_FLAG_BACKUP_SEMANTICS+
+                     FILE_FLAG_OPEN_REPARSE_POINT+FILE_FLAG_OVERLAPPED,
+                     NULL );
     if( h == INVALID_HANDLE_VALUE ) {
```
Comment (guruz) [inline csync/src/vio/csync_vio_local_win.c]: I don't think FILE_FLAG_OVERLAPPED is needed, afaik it's only relevant for the opener (=us)..

File: csync/src/vio/csync_vio_local_win.c [hunk 2]
```diff
@@ -230,8 +230,11 @@ int csync_vio_local_stat(const char *uri, csync_vio_file_stat_t *buf) {
     ULARGE_INTEGER FileIndex;
     mbchar_t *wuri = c_utf8_path_to_locale( uri );
 
-    h = CreateFileW( wuri, 0, FILE_SHARE_READ, NULL, OPEN_EXISTING,
-                     FILE_ATTRIBUTE_NORMAL+FILE_FLAG_BACKUP_SEMANTICS+FILE_FLAG_OPEN_REPARSE_POINT, NULL );
+    h = CreateFileW( wuri, 0, FILE_SHARE_WRITE | FILE_SHARE_READ | FILE_SHARE_DELETE,
```
Comment (ogoffart) [inline csync/src/vio/csync_vio_local_win.c]: Why do you specify the WRITE flag?

**Comment by ogoffart** [comment]
What is this commit fixing?

I'd say "if it is not borken, don't fix it"

**Comment by ckamm** [comment]
Looks good to me.

**Comment by dragotin** [comment]
@ogoffart why shouldn't we allow another process to write the file? I was hoping that this change makes the client a better citizen, and not preventing other programs from doing their job.

**Comment by ogoffart** [comment]
@dragotin your commit try to open the file for writing if i'm not mistaken, which will not work in read only share. Or does it?

**Comment by ogoffart** [comment]
[I deleted the previous comment because i was misunderstanding your commit, confused access flag and shared mode]

So then I have no longer anything against this commit.

However it still is a risky change because it may have undesired side effect (say, when syncing files that are being modified)

**Pull request merged**